Yank/repex.py

```diff
@@ -1201,8 +1259,43 @@ def __init__(self, mcmc_moves=None, number_of_iterations=1, replica_mixing_schem
         # Store constructor parameters. Everything is marked for internal
         # usage because any change to these attribute implies a change
         # in the storage file as well.
+        if number_of_iterations is None:  # Support infinite run
+            number_of_iterations = np.inf
         self._number_of_iterations = number_of_iterations
         self._replica_mixing_scheme = replica_mixing_scheme
+        # Handle Online Analysis
+        if online_analysis_interval is not None and (
+                        type(online_analysis_interval) != int or online_analysis_interval < 1):
+            raise ValueError("online_analysis_interval must be an integer 1 or greater")
+        if online_analysis_interval is not None:
+            if online_analysis_target_error < 0:
+                raise ValueError("online_analysis_target_error must be a float >= 0")
+            elif online_analysis_target_error == 0:
+                logger.warn("online_analysis_target_error of 0 may never converge.")
+            if type(online_analysis_minimum_iterations) is not int or online_analysis_minimum_iterations < 0:
+                raise ValueError("online_analysis_minimum_iterations must be an integer >= 0")
```

Comment:
It may be good to move these checks as part of the relative properties so that they are checked every time they are modified. You could extend the `_StoredProperty` class below to take a `condition` argument of type `function` in `__init__` that check the value and eventually raise the exception. That's a class that helps keeping the options up-to-date with the options in the storage.